model/task/task.go

```diff
@@ -2608,8 +2609,29 @@ func GetTasksByVersion(versionID, sortBy string, statuses []string, variant stri
 		match[DisplayNameKey] = bson.M{"$regex": taskName, "$options": "i"}
 	}
 
+	const tempParentKey = "_parent"
 	pipeline := []bson.M{
 		{"$match": match},
+		// do a self join to filter off execution tasks
+		{"$lookup": bson.M{
+			"from":         Collection,
+			"localField":   IdKey,
+			"foreignField": ExecutionTasksKey,
+			"as":           tempParentKey,
+		}},
```

Comment:
Thanks for doing this.  I think this same step  for filtering also needs to happen for the `countPipeline` portion of this function. So that an accurate count gets returned along side the tasks. 
![image](https://user-images.githubusercontent.com/4605522/102418785-6ed0fb80-3fcc-11eb-93be-6f611b4ccf31.png)

Reply:
I added it to both. Is it possible you're seeing some weird interaction with both my changes and your changes? The counts tests pass